### PR TITLE
Malan tweaks

### PR DESCRIPTION
**Pull request by dmalan**
A few aesthetics and bug fixes, but most importantly, if user logs in with email address, no converts to username (so that cs50.me doesn't think the repo doesn't exist).

### CODE REVIEW

File: opt/cs50/submit50/bin/submit50.py [hunk 1]
```diff
@@ -134,12 +133,17 @@ def authenticate():
     if "X-GitHub-OTP" in res.headers:
         two_factor()
         password = two_factor.token
+
     # check if incorrect password
     elif res.status_code == 401:
         raise Error("Invalid username and/or password.") from None
+
+    # check for other error
```
Comment (dmalan) [inline opt/cs50/submit50/bin/submit50.py]: Know when this might actually happen, @brianyu28?

File: opt/cs50/submit50/bin/submit50.py [hunk 2]
```diff
@@ -134,12 +133,17 @@ def authenticate():
...
     elif res.status_code != 200:
         raise Error("Could not authenticate user.") from None
 
+    # canonicalize (capitalization of) username, especially if user logged in via email address
+    username = res.json()["login"]
```
Comment (dmalan) [inline opt/cs50/submit50/bin/submit50.py]: @glennholloway this addresses the email-to-username issue.

File: opt/cs50/submit50/bin/submit50.py [hunk 3]
```diff
@@ -403,10 +407,16 @@ def run(command, password=None, cwd=None, env=None):
...
 def teardown():
     """Delete temporary directory and temporary file."""
-    spin.keep_spinning = False
-    pexpect.run("rm -rf '{}'".format(run.GIT_DIR))
-    if EXCLUDE:
-        pexpect.run("rm -f '{}'".format(EXCLUDE))
+    if spin.spinning:
+        spin.spinning = False
+        print("\r", end="")
+        sys.stdout.write("\033[K")
```
Comment (dmalan) [inline opt/cs50/submit50/bin/submit50.py]: @glennholloway is this a reliable way of clearing the current line?

File: opt/cs50/submit50/bin/submit50.py (hunk 1 above)
Comment (brianyu28) [inline opt/cs50/submit50/bin/submit50.py]: One that comes to mind right away is GitHub API service outages, which there were a couple of in January.

File: opt/cs50/submit50/bin/submit50.py (hunk 3 above)
Comment (glennholloway) [inline opt/cs50/submit50/bin/submit50.py]: A simpler escape sequence for clearing the current line is `\033[2K`. And "Don't forget to flush!"

Comment (dmalan) [inline opt/cs50/submit50/bin/submit50.py]: thanks! https://github.com/cs50/submit50/commit/9c75e12418db3ef817cbd0a4b503c67b8b31a3f4

**Comment by brianyu28** [comment]
Just tested it, looks good!

**Comment by dmalan** [comment]
Thanks!

**Comment by glennholloway** [comment]
Would you care to give me quick notes for the changelog about the changes that led to 2.1.2 and 2.1.3? I have to bump the version number in `Makefile` anyway.

**Comment by brianyu28** [comment]
Significant changes were:

2.1.2

- add more error handling to `submit50 --checkout` 
- update messages displayed to user to direct them to CS50.me

2.1.3

- canonicalizing username after login to allow login via email
- generalized finding path to python3
- bug fixes

**Comment by glennholloway** [comment]
Much obliged.

**Pull request merged**